src/routes/+error.svelte

```diff
@@ -1,5 +1,81 @@
 <script lang="ts">
+    import { goto } from "$app/navigation";
     import { page } from "$app/state";
+    import { Button } from "$lib/components/primitives/button";
+
+    const errorDetails = $derived(() => {
+        const status = page.status;
+
+        let title: string;
+        let headline: string;
+        let body: string;
+
+        // Set default messages based on common status codes
+        switch (status) {
+            case 401:
+                title = "Unauthorized";
+                headline = "Authentication Required";
+                body = "You need to be logged in to view this page.";
+                break;
+            case 403:
+                title = "Forbidden";
+                headline = "Permission Denied";
+                body = "You do not have the necessary permissions to access this page.";
+                break;
+            case 404:
+                title = "Not Found";
+                headline = "Page Not Found";
+                body = "Sorry, the page you are looking for does not exist.";
+                break;
+            case 500:
+                title = "Server Error";
+                headline = "Internal Server Error";
+                body = "Something went wrong on our end. Our team has been notified.";
```

Comment:
I am not sure, if this is a proper message. It is not even true that our team has been notified right? I would propose something like this:
```suggestion
                body = "A server-side error occurred. Please retry or contact support."
```
or
```suggestion
                body ="The server encountered an error and could not complete your request."
```
But I guess this is a matter of taste. We can discuss about those formulations.